### PR TITLE
Truncate error messages in passportAuth redirectUri

### PR DESCRIPTION
**Pull request by saadaltabari**
Closes: #2831

### What are the changes and their implications?

Error messages representing errors that are thrown while handling the `api/auth/<strategy>/callback` request, will be truncated to a maximum of 100 characters before being passed as to the `?authError=` query  parameter of the redirectUrl. This will ensure that error messages in the request header do not cause the server to refuse processing the redirect request because of its large size.  

## Bug Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

## Feature Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [ ] Documentation added/updated (submit PR to [blitzjs.com repo `canary` branch](https://github.com/blitz-js/blitzjs.com/tree/canary))

### CODE REVIEW

File: packages/blitz-auth/src/server/passport-adapter.ts [hunk 1]
```diff
@@ -161,7 +162,8 @@ export function passportAuth(config: BlitzPassportConfig): ApiHandler {
                 "/"
 
               if (error) {
-                redirectUrl += "?authError=" + encodeURIComponent(error.toString())
+                redirectUrl +=
+                  "?authError=" + encodeURIComponent(truncateString(error.toString(), 100))
```
Comment (beerose) [inline packages/blitz-auth/src/server/passport-adapter.ts:166]: That looks good, but maybe we can additionally console.error the whole message?

**Review by beerose — COMMENTED** [review]
Thanks a lot for this PR! Left one small comment

File: packages/blitz-auth/src/server/passport-adapter.ts (hunk 1 above)
Comment (saadaltabari) [reply packages/blitz-auth/src/server/passport-adapter.ts:168]: done!

**Review by Zeko369 — APPROVED** [review]

File: .changeset/breezy-moose-behave.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"blitz": minor
+"@blitzjs/auth": patch
+---
+
+Error messages representing errors that are thrown while handling the `api/auth/<strategy>/callback` request, will be truncated to a maximum of 100 characters before being passed as to the `?authError=` query parameter of the redirectUrl.
```
Comment (beerose) [inline .changeset/breezy-moose-behave.md:6]:
```suggestion
Truncate errors from `api/auth/<strategy>/callback` request to 100 characters before passing them to the `?authError=` query parameter
```

**Pull request merged**